pyxem/components/scattering_fit_component.py

```diff
@@ -0,0 +1,112 @@
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+"""
+A HyperSpy fitting component to fit the independent atomic scattering
+background to a radial profile.
+
+"""
+import numpy as np
+
+from hyperspy.component import Component
+from diffsims.utils.atomic_scattering_params import ATOMIC_SCATTERING_PARAMS
+from diffsims.utils.lobato_scattering_params import ATOMIC_SCATTERING_PARAMS_LOBATO
+
+
+class ScatteringFitComponent(Component):
+
+    def __init__(self, elements, fracs, N=1., C=0., type='lobato'):
+        """
+        A scattering component to fit background atomic scattering.
+        Calculates the sum of the squares sum_squares = sum (ci * fi**2 )
+        and the square of the sum square_sum = (sum(ci * fi))**2
+        for atomic fraction ci with electron scattering factor fi.
+        The latter is used for normalisation.
+
+        N and C are fitting parameters for the Component class.
+
+        Parameters
+        ----------
+        elements: list of str
+                    A list of elements present (by symbol).
+        fracs: list of float
+                    A list of fraction of the respective elements. Should sum to 1.
+        N : float
+                    The "slope" of the fit.
+        C : float
+                    An additive constant to the fit.
+        type : str
+                    Type of scattering parameters fitted. Default is lobato.
+                    Options are:
+                        - lobato: Fit to Lobato & Van Dyck (2014)
+                        - xtables: Fit to International Tables Vol. C, table 4.3.2.3
+
+        """
+        Component.__init__(self, ['N', 'C'])
+        self.type = type
+        self.elements = elements
+        self.fracs = fracs
+        params = []
+        if self.type == 'lobato':
+            for e in elements:
+                params.append(ATOMIC_SCATTERING_PARAMS_LOBATO[e])
+        elif self.type == 'xtables':
+            for e in elements:
+                params.append(ATOMIC_SCATTERING_PARAMS[e])
+        else:
+            raise NotImplementedError("The parameters `{}` are not implemented."
+                                      "See documentation for available "
+                                      "implementations.".format(type))
+            # As in International Tables for Crystallography, 2006, 4.3.2
+        self.params = params
+
+    def function(self, x):
```

Comment:
I was under the impression that "function" is the necessary name for the base function of a hyperspy component. (hence the name). I will leave a comment.

Reply:
This depends again on whether you can re-write things using the expression functionality of hyperspy. I think we should also consider whether each scattering factor parametrization should be its own component?

I think they probably should be separate components because the function used for the parametrization is different. This way you'd pick which one you want when you decide which component to use for the fit...